write.go

```diff
@@ -28,39 +28,56 @@ var (
 	ErrorUnsupportedBitsPerSample = errors.New("unsupported BitsPerSample value")
 )
 
-// TODO(suyashkumar): consider adding an element-by-element write API.
+// Writer is a struct that allows element-by element writing to a DICOM writer
+type Writer struct {
+	writer dicomio.Writer
+	optSet *writeOptSet
+}
 
-// Write will write the input DICOM dataset to the provided io.Writer as a complete DICOM (including any header
-// information if available).
-func Write(out io.Writer, ds Dataset, opts ...WriteOption) error {
-	optSet := toOptSet(opts...)
+// NewWriter returns a new Writer, that points to the provided io.Writer
+func NewWriter(out io.Writer, opts ...WriteOption) *Writer {
+	optSet := toWriteOptSet(opts...)
 	w := dicomio.NewWriter(out, nil, false)
+
+	return &Writer{
+		writer: w,
+		optSet: optSet,
+	}
+}
+
+// SetTransferSyntax sets the Transfer Syntax for the underlying dicomio.Writer
```

Comment:
nit: end sentence in period here and elsewhere in your docstrings. 

https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences